documentation/smg/14_release_notes.md

```diff
@@ -0,0 +1,57 @@
+# Writing Release Notes
+
+## Release Notes Files
+
+When adding a feature, enhancement or bug fix,
+a developer will update the upcoming release's release notes.
+The release notes documents are held in `documentation/release_notes`.
+There is a release notes file for every major, minor and patch release.
+The documents are ReST files and
+have the same name as the corresponding release, e.g. `v3.5.0.rst`.
+
+## Recording Release Notes
+
+On every pull request, the developer must consider whether the changes require
+documenting.
+If the change will affect users in any way, be it a bug fix, a new feature or
+an API change, then a sentence or two must be added to the latest release notes
+file.
+The release notes are intended to be read by users,
+so the language used should reflect this.
+Use plain English and be brief: try not to exceed 2 sentences.
+
+## Compiling Release Notes
+
+Before the team build/deploy a release,
+the release notes document for that release will be tidied.
+The most significant changes will appear at the top of the document,
```

Comment:
Determined how? Would it be easier to group into sections with each contribution in an arbitrary order, e.g.
- API Changes
- New functionality
- Major bug Fixes (i.e. broken functionality)
- Minor bug fixes (i.e. Changes to user messages)

Reply:
I think this will largely be down to whoever's tidying the notes. API changes are definitely up there, broken (now hopefully fixed) functionality as well. I can stick  _"e.g. API changes"_ on this line, but I feel as though, in the end, it comes down to the judgment of whover's compiling the notes.

Reply:
A template file for people to fill in minimises confusion and means that everyone's on the same page. If you keep it as uniform as possible and leave little room for user judgement, i.e. make it filling in boxes, outside contributors are more likely to do it and you can just point to the template.

Reply:
[There's a template](https://github.com/pace-neutrons/Horace/pull/459/files#diff-255f97fc0e1b86dbeab39bb2bf8b5b015e73f3b63e0ebd4692b95d4fbf57c96a) for us (or external contributors) to fill in as we go (add a bullet point when you open a PR). But before we release, one of us will need to go through whatever's in the document and tidy it up a bit. That's when we should think about which changes need to be emphasized.

Reply:
Sorry, I think I missed your point. Yeah, that's fine. 

I would say that the only tidying should be minimal (formatting and corrections), preferably people should have fleshed them out to the necessary (albeit brief) depth before the PR gets merged because otherwise you will forget the detail and have to revisit it. The "emphasis" should only be moving things around in the same templated block.

Reply:
I disagree on that point.

We want the live changelog to be a quick to update log of all changes that should be included in the final Release Notes.

The critical things with this working document are:
- it's quick and easy to keep up-to-date, so people do it, 
- it includes enough information to get more info later if needed (ie ticket number, developer) plus anything critical that isn't covered elsewhere (e.g. a code fragment that demonstrated the new use-case, or pointer the section of the user docs, if they've updated that).

If you have multiple related or linked changes in a release compiling the RN into a single document lets the author combine those notes into one consistent, clear piece of information for someone reading them.
If you have unrelated things, a few extra words to turn the notes into clear English are needed.

If we have spent a lot of time constructing beautifully formatted sections of documentation here, a lot of that work may need to be redone/discarded in the final edit.